src/generate.mts

```diff
@@ -36,7 +36,7 @@ export async function generate(options: LimitedUserConfig, version: string) {
       asClass: true,
     },
     types: {
-      dates: formattedOptions.useDateType,
+      dates: formattedOptions.useDateType ? "types+transform" : false,
```

Comment:
The type of the `dates` option in `@hey-api/openapi-ts` is `boolean | 'types+transform' | 'types'`.  
Therefore, instead of using a ternary operator here, I believe it would be more appropriate to adjust the type of `LimitedUserConfig`.  

An example of the command when executed via CLI should look as follows:

```bash
openapi-rq -i ./petstore.yaml --useDateType=types+transform
```

Reply:
The `cli.mts` will also look like the following.

```ts
#!/usr/bin/env node
import { readFile } from "node:fs/promises";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { Command, Option } from "commander";
import { defaultOutputPath } from "./constants.mjs";
import { generate } from "./generate.mjs";

const program = new Command();

export type LimitedUserConfig = {
  input: string;
  output: string;
  client?: "angular" | "axios" | "fetch" | "node" | "xhr";
  request?: string;
  format?: "biome" | "prettier";
  lint?: "biome" | "eslint";
  operationId?: boolean;
  serviceResponse?: "body" | "response";
  base?: string;
  enums?: "javascript" | "typescript" | "typescript+namespace";
  useDateType?: boolean | "types+transform" | "types";
  debug?: boolean;
  noSchemas?: boolean;
  schemaType?: "form" | "json";
  pageParam: string;
  nextPageParam: string;
};

const datesType: Extract<
  LimitedUserConfig["useDateType"],
  "types+transform" | "types"
>[] = ["types", "types+transform"];

async function setupProgram() {
 // ...
  program
    .name("openapi-rq")
    .version(version)
    .description("Generate React Query code based on OpenAPI")
    .requiredOption(
      "-i, --input <value>",
      "OpenAPI specification, can be a path, url or string content (required)"
    )
 // ...
    .option(
      "--useDateType [value]",
      "Use Date type instead of string for date types for models. See https://heyapi.vercel.app/openapi-ts/transformers.html#dates",
      (value) => {
        if (value === undefined) {
          return true;
        }
        if (value === "true" || value === "false") {
          return value === "true";
        }
        if (datesType.includes(value as "types+transform" | "types")) {
          return value;
        }

        throw new Error("Invalid value for useDateType");
      },
      false
    )
    // ...
    .parse();

  const options = program.opts<LimitedUserConfig>();
  await generate(options, version);
}

```